dts/bindings/adc/telink,b91-adc.yaml

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2022, Telink Semiconductor
+# SPDX-License-Identifier: Apache-2.0
+
+description: Telink B91 ADC node
+
+compatible: "telink,b91-adc"
+
+include: adc-controller.yaml
+
+properties:
+    reg:
+      required: true
+
+    "#io-channel-cells":
+      const: 1
+
+    sample-freq:
+      type: string
+      required: true
+      enum:
+        - "ADC_SAMPLE_FREQ_23K"
+        - "ADC_SAMPLE_FREQ_48K"
+        - "ADC_SAMPLE_FREQ_96K"
+
+    vref:
+      type: string
+      required: true
+      enum:
+        - "ADC_VREF_0P9V"
+        - "ADC_VREF_1P2V"
```

Comment:
@galak @mbolivar-nordic is this a good practice?

Reply:
@gmarull 
Since the TDS has DT_STRING_TOKEN exactly for this case, I guess it should be OK.
In another case, we do not need DT_STRING_TOKEN at all.
https://docs.zephyrproject.org/2.7.0/reference/devicetree/api.html#c.DT_STRING_TOKEN

Reply:
The problem I see here is HAL driving DT design. Platforms that do not use HALs, or if HAL doesn't have such kind of defines, will likely do something like https://github.com/zephyrproject-rtos/zephyr/pull/46563#discussion_r910717790. I think we should be consistent. Let's wait to see DT maintainers opinions on this topic.

Reply:
@gmarull 
Other vendors also use this approach: e.g. `clock-source` in `nordic,nrf-pdm.yaml`.

Reply:
@gmarull 
The issue with `vref-mv = <900>;` approach is that the only 2 values are exactable, so actually, it is realy `Enum` but not `Int`.
Another potential workaround is to move these parameters to the driver's KConfig and define them as enumeration as well.
But, if the current approach is exactable I prefer to leave it as it is.

Reply:
@gmarull 

Looks like it is the last unresolved comment and in order to speed up the PR, I propose the next 4 options:

1. Accept the current DT implementation. I really like it as it is more user-friendly (in my opinion) because users are informed which values are acceptable. See also all my previous comments for additional explanations why I think the current implementation should be OK.
2. Currently this PR has 14 reviewers including DT maintainers. I propose to close this comment and if somebody from DT maintainers/experts has some DT-related comments/improvements they can create new comments by themselves.
3. We can wait for DT maintainer's response, however, I do not know how much time it will take and the PR will be delayed.
4. I can also easily implement your proposal. Just please confirm it once more and I will do it.

Reply:
Is there some issue with making this an enum instead of a string?

Reply:
So this really should be handled as a `regulator`, but if not then I think we have already in tree usage of `vref-mv` and `zephyr,vref-mv` so we should use on of this in the binding.

Reply:
According to `adc-controller.yaml`, the `zephyr,vref-mv` is dedicated to the external source but not internal, this B91 SoC does not support an external source at all, so I cannot use `vref-mv`.
B91 SoC supports 2 internal sources and this is not supported by the generic `adc-controller.yaml`, and because of that I introduced this new `vref` for B91 SoC to select the internal source (ADC_VREF_0P9V or ADC_VREF_1P2V).

@galak 
Are you OK with the current `telink,b91-adc.yaml` implementation, or should I change something here?

Reply:
The problem here is that Devicetree is being heavily influenced by the HAL. I'd expect for example to specify sample frequency in Hz, e.g. `sample-freq = <DT_FREQ_K(48)>;`, not `sample-freq = "ADC_SAMPLE_FREQ_48K";`. @galak @mbolivar-nordic opinions here?

Reply:
@galak, @mbolivar-nordic, @gmarull 
FYI.. My opinion is: since only 2/3 values are supported, then this is an enumeration but not int (and actually it is related more to the enumeration usage in dts)... and because of that I prefer the current implementation.

Reply:
> @galak, @mbolivar-nordic, @gmarull FYI.. My opinion is: since only 2/3 values are supported, then this is an enumeration but not int (and actually it is related more to the enumeration usage in dts)... and because of that I prefer the current implementation.

you can still enum for `int` properties

Reply:
Oh, got it.. I will rework it, thanks!